src/server/serverinventorymgr.cpp

```diff
@@ -168,6 +168,15 @@ bool ServerInventoryManager::removeDetachedInventory(const std::string &name)
 	return true;
 }
 
+const std::string *ServerInventoryManager::getDetachedInventoryOwner(const std::string &name) const
```

Comment:
`const std::string &`

maybe it'd even make sense to change this into `checkDetachedInvAccess()` and move the checks here

Reply:
`const std::string &` is not ideal for invalid detached inventories (i.e. not found), hence I'll change this to your check function suggestion.

Reply:
Makes sense, I knew there had to be a reason you chose `std::string*`.